update rj_gameplay/README.md (#1863)
RoboJackets/robocup-software: rj_gameplay/README.md

**Implementation - `rj_gameplay/`**

Frequently changing code of gameplay. Subclasses the interfaces and abstract classes in stp/. (Read the docstrings for the relevant superclasses if ever you are confused.)

NOTE: for now, both the old and new infrastructure are existing simultaneously while we transfer over. To figure out if a file is old or new, look at the superclass in the class definition. Old classes subclass an interface (e.g. IPlay, ITactic, ISkill) and new classes subclass an abstract class (e.g. Play, Tactic, Role, Skill).

**Tests - `tests/stp`**

Unit tests for stp/.

## SRTP Framework

See [design doc](https://docs.google.com/document/d/1gRAF--W7FwGoof0--1l_pyjM-N4RmMiV6FhBvyQIRgM/edit?usp=sharing) for more detail.

**Skill - `rj_gameplay/skill/`**

Atomic robot behavior. (e.g. Receive, Move, Kick)

**Role - `rj_gameplay/role/`**

Complex single-robot behavior. (e.g. Goalie, Passer, Receiver)

**Tactic - `rj_gameplay/tactic/`**

Coordinates one or more Roles at a high level. Generates Role Requests. (e.g. PassTactic, WallTactic)

**Play - `rj_gameplay/play/`**

> Coordinates all robots on the field for a given situation. Gets role requests from its tactics, then assigns them according to the cost functions of those role requests. (Examples: Basic122, Basic Defense)

Reviewer: ```suggestion
Coordinates all robots on the field for a given situation. Gets role requests from its tactics, then assigns them according to the cost functions of those role requests. (Examples: Basic Offense, Basic Defense)
```

> The general flow of how a RobotIntent (or specific command for the robot) is created is as follows:

Reviewer 2: ```suggestion
The general flow of how a RobotIntent (or specific command for the robot) is created as follows:
```

> GameplayNode -> SituationAnalyzer -> Play (selected based on situation) -> Tactic(s) -> Role(s) -> Skill(s)

Reviewer: ```suggestion
GameplayNode -> Play Analysis/Selector -> Play (selected based on situation) -> Tactic(s) -> Role(s) -> Skill(s)
```

> Uses SituationAnalyzer to select the best play to run, calling tick() on the

Reviewer: ```suggestion
Uses basic_play_selector to select the best play to run, calling tick() on the
```
